runtime/runtime/src/verifier.rs

```diff
@@ -144,23 +146,45 @@ pub fn verify_and_charge_transaction(
         )?;
     let transaction = &signed_transaction.transaction;
     let signer_id = &transaction.signer_id;
+    let public_key = &transaction.public_key;
 
     let mut signer = match get_account(state_update, signer_id)? {
         Some(signer) => signer,
         None => {
             return Err(InvalidTxError::SignerDoesNotExist { signer_id: signer_id.clone() }.into());
         }
     };
-    let mut access_key = match get_access_key(state_update, signer_id, &transaction.public_key)? {
+    let mut access_key = match get_access_key(state_update, signer_id, public_key)? {
         Some(access_key) => access_key,
-        None => {
-            return Err(InvalidTxError::InvalidAccessKeyError(
-                InvalidAccessKeyError::AccessKeyNotFound {
-                    account_id: signer_id.clone(),
-                    public_key: transaction.public_key.clone(),
-                },
-            )
-            .into());
+        None => match signer_id.get_account_type() {
+            AccountType::EthImplicitAccount => {
+                if derive_account_id_from_public_key(public_key) == *signer_id {
+                    // TODO What about increasing storage usage for that account, because we added access key?
+                    // TODO Can we assume there can only be one access key added to ETH-implicit account, the one from which the address has been derived?
```

Comment:
* Can we remove `SECP256K1` access_key from ETH-implicit and add `ED25519` access key instead?
* Do we assume that transactions from ETH-implicit account are always signed by `SECP256K1` key?
* Can we add public key to the ETH account, other than the one from which the account address has been derived (original public key)? If yes, does the original public key holds the right to sign transactions from this account forever?